search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+let router = express.Router();  // Using Router to divide the app into modules.
+
+const fetch = require('node-fetch'); // Used to access custom search.
+const { Datastore } = require('@google-cloud/datastore');
+const datastore = new Datastore();
+
+const json = require('./../public/country-code.json');
+global.Headers = fetch.Headers;
+
+/** 
+ * Renders a JSON array of the top search results for all countries with API data
+ * obtained every 12 hours for the specified topic.
+ */
+router.get('/:topic', (req, res) => {
+  let topic = req.params.topic;
+  retrieveSearchResultFromDatastore(topic).then(customSearchTopicJsonArray => {
+    // TODO(ntarn): Remove console.log statements when finished debugging.
+    console.log('ntarn debug: In the router.get method'); 
+    console.log(customSearchTopicJsonArray);
+    res.setHeader('Content-Type', 'application/json');
+    res.send(customSearchTopicJsonArray);
+  });
+});
+
+
+/** 
+ * Returns a JSON-formatted array of search results for countries retrieved
+ * from the Datastore.
+ * @param {string} topic Search topic to get data for.
+ */
+async function retrieveSearchResultFromDatastore(topic) {
+  // Request latest entity with a topic matching the given topic.
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp', {
+    descending: true,
+  }).filter('topic', topic).limit(1);
+
+  try {
+    const [customSearchTopic] = await datastore.runQuery(query);
+    let customSearchTopicJsonArray = {
+      topic: customSearchTopic[0].topic,
+      countries: customSearchTopic[0].countries,
+      timestamp: customSearchTopic[0].timestamp,
+    };
+    // TODO(ntarn): Remove console.log statements when finished debugging.
+    console.log(`ntarn debug retrieved search result for topic: ${topic}` +  `example: ${customSearchTopic[0].countries[0].results[0]}`);
+    console.log(customSearchTopicJsonArray);
+    return customSearchTopicJsonArray;
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/** 
+ * Updates daily search results (accumulates by day) in the Datastore.
+ */
+function updateSearchResults() {
+  retrieveGlobalTrends().then(async trends => {
+    for (let i = 0; i < trends.length; i++) { // When testing, make i < 1 trend to test one.
+      await updateSearchResultsForTopic(trends[i].trendTopic);
+      // TODO(ntarn): Remove console.log statements when finished debugging.
+      console.log('ntarn debug updateSearchResult for: ' + trends[i].trendTopic);
+      await new Promise(resolve => setTimeout(resolve, 60000));
+    }
+  });
+}
+
+/** 
+ * Retrieves search result data for all countries for a given topic. Passes
+ * this data to be saved in Datastore.
+ * @param {string} query Search query.
+ */
+async function updateSearchResultsForTopic(query) {
+  let countriesData = [];
+
+  // Note: Can't use forEach with await.
+  for (let i = 0; i < json.length; i++) { // When testing, make i < 3 countries. 
+    var set = 0;
```

Comment:
what's this for?

Reply:
Going through each country to update their search results

Reply:
Yes but what is the variable set for? I also commented and asked why you are using var and not let here below, but I don't think you saw that comment.

Reply:
i deleted the var set already